sos/plugins/cnv.py

```diff
@@ -0,0 +1,234 @@
+    # In future, add a check to run the plugin only when cnv is installed
+    files = (master_cfg,)
+
+    option_list = [
+        ("vm", "collect VM related data, usage: cnv.vm=<vmi name>", '', ""),
+    ]
+
+    def get_resc_ns(self, resc):
+        out = []
+        try:
+            resc_ns_out = open(resc).read()
+        except IOError:
+            return out
+        for line in resc_ns_out.splitlines():
+            if line.startswith("NAMESPACE") \
+                    or line.startswith("No resources found") \
+                    or line.startswith("Error") \
+                    or line.startswith("error"):
+                pass
+            else:
+                out.append(line.split()[1] + " -n " + line.split()[0])
+        return out
+
+    def get_vm_ns(self, vms, option):
+        out = []
+        try:
+            vms_ns_out = open(vms).read()
+        except IOError:
+            return out
+        for line in vms_ns_out.splitlines():
+            if line.startswith("NAMESPACE") \
+                    or line.startswith("No resources found") \
+                    or line.startswith("Error") \
+                    or line.startswith("error"):
+                pass
+            else:
+                if line.split()[1] == option:
+                    out.append(line.split()[1] + " -n " + line.split()[0])
+        return out
+
+    def get_node_info(self, pods):
+        out = ""
+        try:
+            node_out = open(pods).read()
+        except IOError:
+            return out
+        for line in node_out.splitlines():
+            if line.startswith("NAMESPACE") \
+                    or line.startswith("No resources found") \
+                    or line.startswith("Error") \
+                    or line.startswith("error"):
+                pass
+            else:
+                out = line.split()[7]
+        return out
+
+    def setup(self):
+        oc_admin_cfg = os.path.join(self.master_base_dir, "admin.kubeconfig")
+        oc_cmd = "%s --config=%s" % ("oc", oc_admin_cfg)
+
+        all_dep = [
+            "%s get deployment -l kubevirt.io "
+            "--all-namespaces -o wide" % oc_cmd,
+        ]
+        all_svc = [
+            "%s get service -l kubevirt.io "
+            "--all-namespaces -o wide" % oc_cmd,
+            "%s get service -l cdi.kubevirt.io "
+            "--all-namespaces -o wide" % oc_cmd,
+        ]
+        all_pods = [
+            "%s get pods -l kubevirt.io=virt-controller "
+            "--all-namespaces -o wide" % oc_cmd,
+            "%s get pods -l kubevirt.io=virt-api "
+            "--all-namespaces -o wide" % oc_cmd,
+            "%s get pods -l cdi.kubevirt.io "
+            "--all-namespaces -o wide" % oc_cmd,
+            "%s get pods -l app=containerized-data-importer "
+            "--all-namespaces -o wide" % oc_cmd,
+        ]
+        all_stor = [
+            "%s get pvc --all-namespaces -o wide" % oc_cmd,
+        ]
+
+        subcmds = [
```

Comment:
Most of these are either already collected by the kubernetes plugin, or are proposed to be added to it in #1424